### PR TITLE
Homework19 for lesson24

### PR DESCRIPTION
**Pull request by HelensBadCode**

### CODE REVIEW

File: src/homework19/GaragePage/GaragePage.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import BasePage from "../BasePage.js";
+
+
+export default class GaragePage extends BasePage{
+    constructor(page) {
+        super(page, "/panel/garage");
+        this.titleGaragePage =  page.locator('h1', {hasText: 'Garage'});
+        this.profileBtnInHeader = page.getByRole('button', {name: 'My profile'});
```
Comment (sttaran) [inline src/homework19/GaragePage/GaragePage.js]: this button is visible not only on this page

I would create Header component

File: src/homework19/SettingsPage/SettingsPage.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export default class SettingsPage extends BasePage{
+
+    constructor(page) {
+        super(page, '/panel/settings');
+        this.removeUserBlock = page.locator('.user-settings_form', {hasText: 'Remove account'});
+        this.removeUserButton = page.locator('.-remove-account button');
+
+    };
+
+    async removeUserAction(){
```
Comment (sttaran) [inline src/homework19/SettingsPage/SettingsPage.js:13]: return RemoveUserPopup

File: tests/registration_hw19/seccussUserRegistration.spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            const welcomePage = new WelcomePage(page);
+            await welcomePage.navigate();
+            popup = await welcomePage.openSignUpPopup();
+
+
+        })
+
+        test.afterEach(async ({page}) => {
+            const garagePage = new GaragePage(page);
+            const settingsPage = new SettingsPage(page);
```
Comment (sttaran) [inline tests/registration_hw19/seccussUserRegistration.spec.js]: you do not need it

File: tests/registration_hw19/seccussUserRegistration.spec.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            const removeUserPopup = new RemoveUserPopup(page);
+            const sideBar = new SideBarMenu(page);
```
Comment (sttaran) [inline tests/registration_hw19/seccussUserRegistration.spec.js]: do not create ainstance of SideBar in tests, move it to POM

**Review by sttaran — CHANGES_REQUESTED** [review]

File: src/homework19/GaragePage/GaragePage.js (hunk 1 above)
Comment (HelensBadCode) [reply src/homework19/GaragePage/GaragePage.js]: I created Header component

File: src/homework19/SettingsPage/SettingsPage.js (hunk 2 above)
Comment (HelensBadCode) [reply src/homework19/SettingsPage/SettingsPage.js:13]: Added "return RemoveUserPopup"

File: tests/registration_hw19/seccussUserRegistration.spec.js (hunk 4 above)
Comment (HelensBadCode) [reply tests/registration_hw19/seccussUserRegistration.spec.js]: done

File: tests/registration_hw19/seccussUserRegistration.spec.js (hunk 3 above)
Comment (HelensBadCode) [reply tests/registration_hw19/seccussUserRegistration.spec.js]: okey, removed it

File: src/homework19/GaragePage/GaragePage.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+export default class GaragePage extends BasePage{
+    constructor(page) {
+        super(page, "/panel/garage");
+        this.titleGaragePage =  page.locator('h1', {hasText: 'Garage'});
+    };
+
+    async sideBarInstance() {
+        return new SideBarMenu(this._page);
+    }
```
Comment (sttaran) [inline src/homework19/GaragePage/GaragePage.js:14]: not this one

File: src/homework19/GaragePage/GaragePage.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import BasePage from "../BasePage.js";
+import {SideBarMenu} from "./components/SideBarMenu.js";
+import Header from "../../components/Header.js";
+
+
+export default class GaragePage extends BasePage{
+    constructor(page) {
+        super(page, "/panel/garage");
+        this.titleGaragePage =  page.locator('h1', {hasText: 'Garage'});
```
Comment (sttaran) [inline src/homework19/GaragePage/GaragePage.js:9]: do it this way
`this.sideBar = new SideBarMenu(this._page)`

File: src/homework19/GaragePage/GaragePage.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    };
+
+    async sideBarInstance() {
+        return new SideBarMenu(this._page);
+    }
+
+    async presentProfile(){
+       return new Header(this._page);
+    }
```
Comment (sttaran) [inline src/homework19/GaragePage/GaragePage.js:18]: same here
 move it to constructor

**Review by sttaran — CHANGES_REQUESTED** [review]

File: src/homework19/GaragePage/GaragePage.js (hunk 5 above)
Comment (HelensBadCode) [reply src/homework19/GaragePage/GaragePage.js:14]: fixed in https://github.com/HelensBadCode/AQA_playwright/pull/4/commits/6a148f556810e850b7c928855bcf90190e00fe0f

File: src/homework19/GaragePage/GaragePage.js (hunk 6 above)
Comment (HelensBadCode) [reply src/homework19/GaragePage/GaragePage.js:9]: done)

File: src/homework19/GaragePage/GaragePage.js (hunk 7 above)
Comment (HelensBadCode) [reply src/homework19/GaragePage/GaragePage.js:18]: also done)

**Pull request merged**